### PR TITLE
fix update shard config bug

### PR DESCRIPTION
**Pull request by qiluge**

### CODE REVIEW

File: core/store/ledgerstore/ledger_store.go [hunk 1]
```diff
@@ -173,18 +173,20 @@ func (this *LedgerStoreImp) InitLedgerStoreWithGenesisBlock(genesisBlock *types.
...
-			return fmt.Errorf("HashBlockExist error %s", err)
-		}
-		if !exist {
-			return fmt.Errorf("GenesisBlock arenot init correctly, hash %s", genesisHash.ToHexString())
-		}
 		err = this.init()
 		if err != nil {
 			return fmt.Errorf("init error %s", err)
 		}
+		if this.currBlockHeight == 0 {
```
Comment (Honglei-Cong) [inline core/store/ledgerstore/ledger_store.go:180]: remove this if?
the following checks are always needed.

File: core/chainmgr/chainmgr.go [hunk 2]
```diff
@@ -196,7 +196,7 @@ func (self *ChainManager) initMainLedger(stateHashHeight uint32) error {
 }
 
 func (self *ChainManager) initShardLedger(shardInfo *ShardInfo) error {
-	if shardInfo != nil && ledger.GetShardLedger(shardInfo.ShardID) != nil {
+	if ledger.GetShardLedger(shardInfo.ShardID) != nil {
```
Comment (Honglei-Cong) [inline core/chainmgr/chainmgr.go:199]: necessary change here?

Comment (qiluge) [reply core/chainmgr/chainmgr.go:199]: If we don't remove it, we need to judge ```shardInfo == nil ``` in addition

File: core/store/ledgerstore/ledger_store.go (hunk 1 above)
Comment (qiluge) [reply core/store/ledgerstore/ledger_store.go:180]: This is a temporary change, which will be optimized later

**Pull request merged**